From: Mark Fasheh <mfasheh@suse.com>
Patch-mainline: 2.6.28
Subject: Remove un-defined acl handlers 

These handlers aren't actually defined yet. Having them around causes more
trouble than it's worth. The ACL support patches will add them in a proper
fashion.

Signed-off-by: Mark Fasheh <mfasheh@suse.com>

diff -pruN suse/trees/linux-2.6.27-ocfs2/fs/ocfs2//xattr.c linux-2.6.git/fs/ocfs2//xattr.c
--- suse/trees/linux-2.6.27-ocfs2/fs/ocfs2/xattr.c	2009-02-05 15:46:43.000000000 -0800
+++ linux-2.6.git/fs/ocfs2/xattr.c	2009-02-05 15:51:33.000000000 -0800
@@ -79,29 +79,13 @@ static struct ocfs2_xattr_def_value_root
 
 struct xattr_handler *ocfs2_xattr_handlers[] = {
 	&ocfs2_xattr_user_handler,
-#ifdef CONFIG_OCFS2_FS_POSIX_ACL
-	&ocfs2_xattr_acl_access_handler,
-	&ocfs2_xattr_acl_default_handler,
-#endif
 	&ocfs2_xattr_trusted_handler,
-#ifdef CONFIG_OCFS2_FS_SECURITY
-	&ocfs2_xattr_security_handler,
-#endif
 	NULL
 };
 
 static struct xattr_handler *ocfs2_xattr_handler_map[] = {
 	[OCFS2_XATTR_INDEX_USER]	= &ocfs2_xattr_user_handler,
-#ifdef CONFIG_OCFS2_FS_POSIX_ACL
-	[OCFS2_XATTR_INDEX_POSIX_ACL_ACCESS]
-					= &ocfs2_xattr_acl_access_handler,
-	[OCFS2_XATTR_INDEX_POSIX_ACL_DEFAULT]
-					= &ocfs2_xattr_acl_default_handler,
-#endif
 	[OCFS2_XATTR_INDEX_TRUSTED]	= &ocfs2_xattr_trusted_handler,
-#ifdef CONFIG_OCFS2_FS_SECURITY
-	[OCFS2_XATTR_INDEX_SECURITY]	= &ocfs2_xattr_security_handler,
-#endif
 };
 
 struct ocfs2_xattr_info {
diff -pruN suse/trees/linux-2.6.27-ocfs2/fs/ocfs2//xattr.h linux-2.6.git/fs/ocfs2//xattr.h
--- suse/trees/linux-2.6.27-ocfs2/fs/ocfs2/xattr.h	2009-02-05 15:46:43.000000000 -0800
+++ linux-2.6.git/fs/ocfs2/xattr.h	2009-02-05 15:51:33.000000000 -0800
@@ -40,13 +40,6 @@ enum ocfs2_xattr_type {
 
 extern struct xattr_handler ocfs2_xattr_user_handler;
 extern struct xattr_handler ocfs2_xattr_trusted_handler;
-#ifdef CONFIG_OCFS2_FS_POSIX_ACL
-extern struct xattr_handler ocfs2_xattr_acl_access_handler;
-extern struct xattr_handler ocfs2_xattr_acl_default_handler;
-#endif
-#ifdef CONFIG_OCFS2_FS_SECURITY
-extern struct xattr_handler ocfs2_xattr_security_handler;
-#endif
 
 extern ssize_t ocfs2_listxattr(struct dentry *, char *, size_t);
 extern int ocfs2_xattr_get(struct inode *, int, const char *, void *, size_t);
